src/irr/asset/CGLSLVirtualTexturingBuiltinIncludeLoader.h

```diff
@@ -0,0 +1,368 @@
+#ifndef _IRR_C_GLSL_VIRTUAL_TEXTURING_BUILTIN_INCLUDE_LOADER_H_INCLUDED__
+#define _IRR_C_GLSL_VIRTUAL_TEXTURING_BUILTIN_INCLUDE_LOADER_H_INCLUDED__
+
+#include "irr/asset/CGraphicsPipelineLoaderMTL.h"
+
+namespace irr {
+namespace asset
+{
+
+class CGLSLVirtualTexturingBuiltinIncludeLoader : public IBuiltinIncludeLoader
+{
+public:
+    const char* getVirtualDirectoryName() const override { return "glsl/virtual_texturing/"; }
+
+private:
+	static core::vector<std::string> parseArgumentsFromPath(const std::string& _path)
+	{
+		core::vector<std::string> args;
+
+		std::stringstream ss{ _path };
+		std::string arg;
+		while (std::getline(ss, arg, '/'))
+			args.push_back(std::move(arg));
+
+		return args;
+	}
+	static std::string getDescriptors(const std::string& _path)
+	{
+		auto args = parseArgumentsFromPath(_path.substr(_path.find_first_of('/') + 1, _path.npos));
+		constexpr uint32_t
+			set_ix		= 0u,
+			pgt_bnd_ix	= 1u,
+			f_bnd_ix	= 2u,
+			i_bnd_ix	= 3u,
+			u_bnd_ix	= 4u,
+			f_cnt_ix	= 5u,
+			i_cnt_ix	= 6u,
+			u_cnt_ix	= 7u;
+
+		return 
+"\n#ifndef _IRR_VT_DESCRIPTOR_SET"
+"\n#define _IRR_VT_DESCRIPTOR_SET " + args[set_ix] +
+"\n#endif" +
+"\n#ifndef _IRR_VT_PAGE_TABLE_BINDING"
+"\n#define _IRR_VT_PAGE_TABLE_BINDING " + args[pgt_bnd_ix] +
+"\n#endif" +
+"\n#ifndef _IRR_VT_FLOAT_VIEWS"
+"\n#define _IRR_VT_FLOAT_VIEWS_BINDING " + args[f_bnd_ix] +
+"\n#define _IRR_VT_FLOAT_VIEWS_COUNT " + args[f_cnt_ix] +
+"\n#endif" +
+"\n#ifndef _IRR_VT_INT_VIEWS"
+"\n#define _IRR_VT_INT_VIEWS_BINDING " + args[i_bnd_ix] +
+"\n#define _IRR_VT_INT_VIEWS_COUNT " + args[i_cnt_ix] +
+"\n#endif" +
+"\n#ifndef _IRR_VT_UINT_VIEWS"
+"\n#define _IRR_VT_UINT_VIEWS_BINDING " + args[u_bnd_ix] +
+"\n#define _IRR_VT_UINT_VIEWS_COUNT " + args[u_cnt_ix] +
+"\n#endif" +
+R"(
+layout(set=_IRR_VT_DESCRIPTOR_SET, binding=_IRR_VT_PAGE_TABLE_BINDING) uniform usampler2DArray pageTable;
+#if _IRR_VT_FLOAT_VIEWS_COUNT
+layout(set=_IRR_VT_DESCRIPTOR_SET, binding=_IRR_VT_FLOAT_VIEWS_BINDING) uniform sampler2DArray physicalTileStorageFormatView[_IRR_VT_FLOAT_VIEWS_COUNT];
+#endif
+#if _IRR_VT_INT_VIEWS_COUNT
+layout(set=_IRR_VT_DESCRIPTOR_SET, binding=_IRR_VT_INT_VIEWS_BINDING) uniform isampler2DArray iphysicalTileStorageFormatView[_IRR_VT_INT_VIEWS_COUNT];
+#endif
+#if _IRR_VT_UINT_VIEWS_COUNT
+layout(set=_IRR_VT_DESCRIPTOR_SET, binding=_IRR_VT_UINT_VIEWS_BINDING) uniform usampler2DArray uphysicalTileStorageFormatView[_IRR_VT_UINT_VIEWS_COUNT];
+#endif
+)";
+	}
+	static std::string getExtensions(const std::string&)
+	{
+		return R"(
+#extension GL_EXT_nonuniform_qualifier  : enable
+
+//tmp dirty fix for weird behavior of renderdoc
+//uncomment below if having troubles in renderdoc and you're sure GL_NV_gpu_shader5 is available
+#define RUNNING_IN_RENDERDOC
+#ifdef RUNNING_IN_RENDERDOC
+	#define IRR_GL_NV_gpu_shader5
+#endif
```

Comment:
I need you to fix this and read the extension list in `COpenGLDriver` from env-var if such is set

Reply:
https://github.com/buildaworldnet/IrrlichtBAW/pull/474#discussion_r416053604

Reply:
oh and you could inject a `IRR_GL_IRR_RUNNING_IN_RENDERDOC` because the `COpenGLDriver` can detect the Renderdoc.dll

Reply:
hm, instead of this envvar?

Reply:
no both, to be done

Regular OpenGL extensions can come from envvar or not
We define extra extension on top of all that to let GLSL know its running in renderdoc